R/arc-select.R

```diff
@@ -134,7 +172,10 @@ collect_layer <- function(x, n_max = Inf, token = Sys.getenv("ARCGIS_TOKEN"), ..
 
   # stop if query not supported
   if (!grepl("query", x[["capabilities"]], ignore.case = TRUE)) {
-    stop("Feature Layer ", x[['name']], " does not support querying")
+    cli::cli_abort(
+      "Feature Layer ", x[['name']], " does not support querying",
```

Comment:
We need to check the class here instead of `"Feature Layer"`. I know this isn't your code but its included in a diff so its yours now!!! 

We could theoretically, apply a query to an `ImageServer` even, so this need to report the class of the object instead of assuming FeatureLayer